src/apps/api/tests/test_queues.py

```diff
@@ -0,0 +1,177 @@
+        self.client.login(username='test', password='test')
+        response, _ = self.queue_api_request_with_mock(
+            'put',
+            'queues-detail',
+            {'pk': self.created_queue.id},
+            self._get_test_update_data()
+        )
+        assert response.status_code == 200
+        assert Queue.objects.get(pk=self.created_queue.id).name == 'A BRAND NEW NAME'
+
+        response, _ = self.queue_api_request_with_mock(
+            'delete',
+            'queues-detail',
+            {'pk': self.created_queue.id},
+            {}
+        )
+        assert response.status_code == 204
+        assert not Queue.objects.filter(pk=self.created_queue.id).exists()
+
+    def test_other_user_cannot_delete_or_edit_queue(self):
+        self.client.login(username='another_user', password='another_user')
+        response, _ = self.queue_api_request_with_mock(
+            'put',
+            'queues-detail',
+            {'pk': self.created_queue.id},
+            self._get_test_update_data()
+        )
+        assert response.status_code == 403
+        assert Queue.objects.get(pk=self.created_queue.id).name != 'A BRAND NEW NAME'
+
+        response, _ = self.queue_api_request_with_mock(
+            'delete',
+            'queues-detail',
+            {'pk': self.created_queue.id},
+            {}
+        )
+
+        assert response.status_code == 403
+        assert Queue.objects.filter(pk=self.created_queue.id).exists()
+
+    def test_collab_user_cannot_delete_or_edit_queue(self):
+        self.client.login(username='organizer_user', password='organizer_user')
+        response, _ = self.queue_api_request_with_mock(
+            'put',
+            'queues-detail',
+            {'pk': self.created_queue.id},
+            self._get_test_update_data()
+        )
+        assert response.status_code == 403
+        assert Queue.objects.get(pk=self.created_queue.id).name != 'A BRAND NEW NAME'
+
+        response, _ = self.queue_api_request_with_mock(
+            'delete',
+            'queues-detail',
+            {'pk': self.created_queue.id},
+            {}
+        )
+
+        assert response.status_code == 403
+        assert Queue.objects.filter(pk=self.created_queue.id).exists()
+
+    # --------- Queue Limit Tests ---------
+
+    # Test user queue limits
+    def test_queue_limits_for_regular_users(self):
+        self.user.rabbitmq_queue_limit = 1
+        response, _ = self.queue_api_request_with_mock(
+            'post',
+            'queues-list',
+            {},
+            self._get_test_update_data(return_id=False)
+        )
+        assert response.status_code == 403
+        assert self.user.rabbitmq_queue_limit == self.user.queues.count()
+
+    # Test super user queue limits
+    def test_super_users_has_no_queue_limit(self):
+        # Make test user a superuser before hand
+        self.user.is_superuser = True
+        self.user.rabbitmq_queue_limit = 2
```

Comment:
you can set these values in the UserFactory, so you don't need to do it in the tests